crates/bls-snark-sys/src/signatures.rs

```diff
@@ -144,18 +168,21 @@ pub extern "C" fn hash_composite_cip22(
     in_extra_data_len: c_int,
     out_hash: *mut *mut u8,
     out_len: *mut c_int,
+    attempt_counter: *mut u8,
 ) -> bool {
     convert_result_to_bool::<_, BLSError, _>(|| {
         let message = unsafe { slice::from_raw_parts(in_message, in_message_len as usize) };
         let extra_data =
             unsafe { slice::from_raw_parts(in_extra_data, in_extra_data_len as usize) };
-        let hash = COMPOSITE_HASH_TO_G1_CIP22.hash(SIG_DOMAIN, message, extra_data)?;
+        let mut counter = 0;
+        let hash = COMPOSITE_HASH_TO_G1_CIP22.hash_with_counter(SIG_DOMAIN, message, extra_data, &mut counter)?;
```

Comment:
Same comment as above - I think you can use `hash_with_attempt` and get the counter from the return value. That's what we do in the SNARK.